applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -266,102 +442,27 @@ static struct usb_hid_device *subscriber_to_usb_hid(const void *subscriber)
 	return NULL;
 }
 
-static bool is_hid_boot_report(uint8_t report_id)
-{
-	return ((report_id == REPORT_ID_BOOT_MOUSE) || (report_id == REPORT_ID_BOOT_KEYBOARD));
-}
-
-static bool is_hid_boot_report_supported(uint8_t report_id)
+static bool handle_hid_report_event(struct hid_report_event *event)
 {
-	__ASSERT_NO_MSG(is_hid_boot_report(report_id));
-
-	return ((IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE) &&
-		 (report_id == REPORT_ID_BOOT_MOUSE)) ||
-		(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD) &&
-		 (report_id == REPORT_ID_BOOT_KEYBOARD)));
-}
+	struct usb_hid_device *usb_hid = subscriber_to_usb_hid(event->subscriber);
 
-static bool can_send_hid_report(struct usb_hid_device *usb_hid, uint8_t report_id)
-{
-	if (!usb_hid->enabled) {
-		LOG_WRN("Cannot send report: USB not connected");
+	if (!usb_hid) {
+		/* Not us. */
 		return false;
 	}
 
-	if (usb_hid->hid_protocol == HID_PROTOCOL_BOOT) {
-		if (!is_hid_boot_report(report_id)) {
-			LOG_WRN("Cannot send report: incompatible boot/report mode");
-			return false;
-		}
+	const uint8_t *data = event->dyndata.data;
+	size_t size = event->dyndata.size;
+	struct usb_hid_buf *sending_buf = usb_hid_buf_find(usb_hid, USB_HID_BUF_SENDING);
```

Comment:
As a follow-up, you can also add an assert to check ALLOCATION status for sending_buf

Reply:
Added. Keep in mind that it would be checked by `report_sent` soon anyway.